src/Auth0.php

```diff
@@ -510,17 +225,18 @@ public function exchange(): bool
         }
 
         if (isset($response['id_token'])) {
-            if (! $this->transientHandler->isset('nonce')) {
-                throw new \Auth0\SDK\Exception\InvalidTokenException('Nonce value not found in application store');
+            if (! $this->transient->isset('nonce')) {
+                throw \Auth0\SDK\Exception\StateException::missingNonce();
             }
 
             $this->setIdToken($response['id_token']);
         }
 
-        if ($this->skipUserinfo) {
-            $user = $this->idTokenDecoded;
-        } else {
-            $user = $this->authentication->userInfo($this->accessToken);
+        $user = $this->state->getIdTokenDecoded();
```

Comment:
Not for this PR, but would be nice to surface the access token expiry from the code exchange, so user's can make sensible decisions about when to refresh the Access Token

eg https://github.com/auth0/auth0-PHP/pull/425/files#diff-fdd37373daf341f412bc52cc50c1e06d26291383643f8737c79d4e125e735ea2R584

Reply:
Good call there, I'll figure out a way of exposing this, perhaps in the new SdkState class. This would be good to make accessible for sure.